### DISCUSSION TITLE
GRPC

### DISCUSSION THREAD

**Discussion by MNouroozi** [cypress-io/cypress#19653] [Questions and Help]
hi,How Can test grpc Poroto in cypress?
how plz?

**Comment by mariuszadam** [comment]
Hello @MNouroozi 

I was able to test basic "hello world" grpc service using cy.exec() and grpcC
https://www.npmjs.com/package/grpcc

with cy.exec()

and it looks like this:

`
let cmd = "grpcc --proto ./check.proto --address localhost:3000 --insecure --eval \'client.methodName({\"service\": \"Hello World\"}, printReply)\' ";


    cy.exec(
        cmd
    ).its('stdout')
     .should('contain', 'HELLO')
     
     
         cy.exec(
        cmd
    ).then(result => {
        // yields the 'result' object, result object components are yielded as strings
        expect(result.code).to.equal(0);

        // result.stdout["status"] is undefined
        expect(result.stdout).to.include("HELLO");

        let jsonResultStdOut = JSON.parse(result.stdout);
        // @ts-ignore
        expect(jsonResultStdOut["status"]).to.equal("HELLO");
    })
`

**Reply by idanElitzur** [reply]
Can this ability be a new fature in the next releases as part of the `cy.network()` options?
@jennifer-shehane
